bam_markdup.c

```diff
@@ -635,10 +639,42 @@ static void make_single_key(key_data_t *key, bam1_t *bam) {
         orientation = O_FF;
     }
 
+    if (param->barcode) {
+        if ((data = bam_aux_get(bam, param->barcode))) {
+            if (!(bar = bam_aux2Z(data))) {
+                fprintf(stderr, "[markdup] error: %s tag wrong type. Aux tag needs to be a string type.\n", param->barcode);
+                // return 1; FIXME, error checking on single keys
+            }
+
+            barcode = do_hash((unsigned char *)bar, strlen(bar));
```

Comment:
Yes, the error handling on this and the paired keys needs to be redone.

Reply:
I have removed the error handling in both key making functions.  If the there is mixed barcode/non-barcode data then having an error would be the wrong thing to do.  I will add barcode counts to the stats in my next piece of markdup work.

Reply:
I don't think that change went the right way.  If either `bam_aux2Z` or `regexec` fail it needs to set `barcode` to some default value (maybe 0?) instead of carrying on an trying to use values that haven't been set correctly.

Reply:
Ahhgh, I fixed the regex bits but not the aux tag sections.

Reply:
Ah, but the `regexec` fix also wasn't right because it now it ignores the error and tries to use `matches[1]` which may or may not have been set to something.  Instead it needs to check the`regexec` return value, and skip everything else if it didn't work.

Reply:
What?  Where are you seeing that?  `barcode` never gets set to `matches[1]`.